staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -7645,3 +7659,19 @@ const (
 	// the destination set to the node's IP and port or the pod's IP and port.
 	LoadBalancerIPModeProxy LoadBalancerIPMode = "Proxy"
 )
+
+// ImageVolumeSource represents a image volume resource.
+type ImageVolumeSource struct {
+	// Required: Image or artifact reference to be used.
+	// Behaves in the same way as pod.spec.containers[*].image.
+	// More info: https://kubernetes.io/docs/concepts/containers/images
+	Reference string `json:"reference" protobuf:"bytes,1,opt,name=reference"`
+
+	// Policy for pulling OCI objects. Possible values are:
+	// Always: the kubelet always attempts to pull the reference. Container creation will fail If the pull fails.
+	// Never: the kubelet never pulls the reference and only uses a local image or artifact. Container creation will fail if the reference isn't present.
+	// IfNotPresent: the kubelet pulls if the reference isn't already present on disk. Container creation will fail if the reference isn't present and the pull fails.
+	// Defaults to IfNotPresent
```

Comment:
A point of inconsistency with containers image - the pull policy for a container is overriden to "Always" if a latest tag is used (i.e. `// Defaults to Always if :latest tag is specified, or IfNotPresent otherwise.`).

While this behavior is confusing, I think it would be more confusing if we were not consistent with the container image field in this fashion.  We would need a really strong argument to *not* be consistent - while I tried to find one (I don't like the behavior), I could not generate one that was strong enough to create a point of inconsistency.

For that reason, I think we need to have the same behavior and the same godoc here.

Reply:
yeah, agree.

Reply:
Makes sense, I updated the docs wrt that detail.